### DISCUSSION TITLE
Support for klog in skid (assignment scheduler)

### DISCUSSION THREAD

**Discussion by acikek** [jotaen/klog#98] [Show and tell]
Hey all, been awhile :) I've been working on this project recently called `skid` that tracks school assignments for students. Seeing as assignments are organized into dates (records) and classes (tags), I thought support for klog would be a perfect addition. So, I pushed that just now!

In my tool, this is pretty useful for assignments you're just about to mark as 'complete', since no other information other than completed assignments' names gets pushed to that section. Unfortunately, there's no data for how long an assignment might take, so I just have the user input an average for all the assignments, such as `1h`, and they can modify it later if need be.

Here's an example, with some test assignments in my chemistry / Spanish classes:
![Screenshot_20210829_115906](https://user-images.githubusercontent.com/26909391/131262259-44329564-ba3d-4b25-8688-be7ff4129884.png)

You can check the project out on crates.io [here](https://crates.io/crates/skid), and the repo is [here](https://github.com/acikek/skid). Enjoy!

**Comment by jotaen** [comment]
Cool idea to integrate klog that way into your tool, it seems like a lightweight addition. Just out of curiosity, how do you store the data? Is in in plain text or do you use a database for example?

**Reply by acikek** [reply]
It's a custom format. You can use the `encode` command to see what the data will look like. 
The format is like this: each line is a class, sorted by period. `class_id,Class Name,[Assignment Name;d-m-y]...,completed assignments...`

Here is my current file:
```
chm,Chemistry Honors,1,density problems,balancing problems,scientific notation problems
csa,AP Computer Science A,2
mah,Math Analysis Honors,3,p305 39-59 odd
eng,English III,4,[bury my heart at a wounded knee;08-09-2021],writing plan,narrative draft,of plymouth plantation
esp,Spanish III,5,traer pizarra,escribe números
ush,US History,6,[cultural expression;07-09-2021],thinking about history,native american statistics,a different mirror
```
